boofuzz/sessions.py

```diff
@@ -660,6 +670,9 @@ def _main_fuzz_loop(self, fuzz_case_iterator):
             self._fuzz_data_logger.log_error("Restarting the target failed, exiting.")
             self.export_file()
             raise
+        except exception.BoofuzzCallbackError:
+            self._fuzz_data_logger.log_error("Custom callback function raised uncaught exception, exiting.")
```

Comment:
Will change

Reply:
I just noticed that this would require `exception.BoofuzzCallbackError` to take the name of the failing callback as an argument. Not that this would be a problem but if we do that we will get the same line twice in the log.
This is how it currently looks like:
```
[2019-03-04 17:49:56,334] Test Case: 1: Test.verbs.1
[2019-03-04 17:49:56,334]     Info: Type: Group. Default value: 'GET'. Case 1 of 384 overall.
[2019-03-04 17:49:56,334]     Info: Opening target connection (127.0.0.1:80)...
[2019-03-04 17:49:56,334]     Info: Connection opened.
[2019-03-04 17:49:56,334]   Test Step: Fuzzing Node 'Test'
[2019-03-04 17:49:56,335]     Info: Sending 8 bytes...
[2019-03-04 17:49:56,336]     Transmitted 8 bytes: 47 45 54 20 0d 0a 0d 0a 'GET \r\n\r\n'
[2019-03-04 17:49:56,336]     Info: Closing target connection...
[2019-03-04 17:49:56,336]     Info: Connection closed.
[2019-03-04 17:49:56,336]   Test Step: Post- test case callback: "ping"
[2019-03-04 17:49:56,336]     Info: Ping Pong
[2019-03-04 17:49:56,337]     Error!!!! A custom post_send callback function raised an uncought error. Stopping test run.
                              Traceback (most recent call last):
                                File "/home/dev/PycharmProjects/boofuzz/boofuzz/sessions.py", line 1412, in _post_send
                                  f(target=target, fuzz_data_logger=self._fuzz_data_logger, session=self, sock=target)
                                File "/home/dev/PycharmProjects/boofuzz/requests/http.py", line 17, in ping
                                  a = 5 / 0
                              ZeroDivisionError: integer division or modulo by zero
                              
[2019-03-04 17:49:56,338]   Test Step: Cleaning up connections from callbacks
[2019-03-04 17:49:56,338]     Info: Closing target connection...
[2019-03-04 17:49:56,338]     Info: Connection closed.
[2019-03-04 17:49:56,339]     Error!!!! Custom callback function raised uncaught exception, exiting.
```
Using the `ERR_CALLBACK_FUNC`, the last line would change to
```
[2019-03-04 17:49:56,339]     Error!!!! A custom post_send callback function raised an uncought error. Stopping test run.
```
If we keep the current one, I guess `Error!!!! A custom callback function raised an uncaught exception, exiting.` would be a little better, but that's all up to you.

Reply:
OK, I didn't think about that complexity of passing the information up the exception chain. Looks good as is.